sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/xml/XSDToSchema.scala

```diff
@@ -48,7 +49,7 @@ object XSDToSchema extends Logging{
     val in = ValidatorUtil.openSchemaFile(xsdPath)
     val xmlSchemaCollection = new XmlSchemaCollection()
     xmlSchemaCollection.setBaseUri(xsdPath.toString)
-    val xmlSchema = xmlSchemaCollection.read(new InputStreamReader(in))
+    val xmlSchema = xmlSchemaCollection.read(new InputStreamReader(in, StandardCharsets.UTF_8))
```

Comment:
This is arguable because if you don't specify the encoding, then it will pick the system default encoding up. If your XSD file is written in other encodings, but here tries to read UTF-8, it will fail.

Other places they have to be UTF-8 because Spark encodes so explicitly.

Reply:
Thanks，I cite hive firstly，than feedback to this pr? @HyukjinKwon

Reply:
As described above, I wouldn't touch Hive side.

Reply:
XSDtoSchema would not modify it， than HiveImpl had also changed can refer recent pr: https://github.com/apache/hive/pull/5243